src/components/Modeler.vue

```diff
@@ -586,6 +587,42 @@ export default {
         }
       });
     },
+    addCancelActivityToBoundaryEvents() {
```

Comment:
I think there's an awful lot going on in this method and I'm not sure if the code really reveals its intention that well. Could we refactor this, perhaps into multiple methods, so that the code is clearer?